File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
/*创建控制层*/
app.controller("brandController",function ($scope,$http,$controller,brandService) {
    $controller("baseController",{$scope:$scope});
    $scope.findAll = function(){
        brandService.findAll().success(function (response) {
            $scope.list = response;
        });
    };

    //分页
    $scope.findPage = function (pageNum, pageSize) {
        brandService.findPage(pageNum, pageSize).success(function (response) {
            $scope.paginationConf.totalItems = response.total;
            $scope.list = response.rows;
        })
    };
    //添加
    $scope.save = function(){
        var object = null;
        if($scope.entity.id!=null){
            object = brandService.update($scope.entity)
        }else {
            object = brandService.add($scope.entity)
        }
        object.success(function (response) {
            if(response.success){
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        })
    };
    //查找一个实体回显
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        });
    };

    /*删除*/
    $scope.del = function () {
        brandService.del($scope.selectIds).success(function (response) {
            if(response.success){
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        })
    };
    $scope.searchEntity = {}; // 初始化加载的时候，给它赋值一个空的对象
    $scope.search = function (pageNum, pageSize) {
        brandService.search(pageNum, pageSize,$scope.searchEntity).success(function (response) {
            $scope.paginationConf.totalItems = response.total;
            $scope.list = response.rows;
        })
    };
});

File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.github.pagehelper.Page;
import com.github.pagehelper.PageHelper;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * brand
 */
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(Integer pageNum,Integer pageSize){
        return brandService.findAll();
    }
    @RequestMapping("/pageFind")
    public PageResult pageFind(Integer pageNum,Integer pageSize){
        return brandService.pageFind(pageNum,pageSize);
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){
        try {
            brandService.add(tbBrand);
            return new Result(true,"增加成功");
        } catch (Exception e) {
            return new Result(false,"增加失败");
        }
    }
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            for (Long id : ids) {
                brandService.delete(id);
            }
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand, Integer pageNum,Integer pageSize){
        return brandService.findPage(tbBrand,pageNum,pageSize);
    }
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}
